src/readiness/statusManager.js

```diff
@@ -1,34 +1,35 @@
+import promiseWrapper from '../utils/promise/wrapper';
 import logFactory from '../utils/logger';
 const log = logFactory('');
 
 const NEW_LISTENER_EVENT = 'newListener';
 const REMOVE_LISTENER_EVENT = 'removeListener';
 
-export default function callbackHandlerContext(context, forSharedClient = false) {
+// default onRejected handler, that just logs the error, if ready promise doesn't have one.
+function defaultOnRejected(err) {
+  log.error(err);
+}
+
+/**
+ * StatusManager factory.
+ * Responsable of exposing public status API: ready promise, event emitter and constants (SDK_READY, etc).
+ * It also updates client context according to status events and logs related warnings and errors.
+ *
+ * @param {Object} context client context
+ * @param {number} internalReadyCbCount number of SDK_READY listeners that are added/removed by the SDK,
+ * used to properly log the warning 'No listeners for SDK Readiness detected'
+ */
+export default function callbackHandlerContext(context, internalReadyCbCount = 0) {
   const gate = context.get(context.constants.READINESS).gate;
   let readyCbCount = 0;
   let isReady = false;
+  let hasTimedout = false;
   const {
     SDK_READY,
     SDK_READY_FROM_CACHE,
     SDK_UPDATE,
     SDK_READY_TIMED_OUT
   } = gate;
-  const readyPromise = getReadyPromise();
-
-  gate.once(SDK_READY, () => {
-    if (readyCbCount === 0) log.warn('No listeners for SDK Readiness detected. Incorrect control treatments could have been logged if you called getTreatment/s while the SDK was not yet ready.');
-
-    context.put(context.constants.READY, true);
-
-    isReady = true;
-  });
-
-  gate.once(SDK_READY_FROM_CACHE, () => {
-    log.info('Split SDK is ready from cache.');
-
-    context.put(context.constants.READY_FROM_CACHE, true);
-  });
 
   gate.on(REMOVE_LISTENER_EVENT, event => {
```

Comment:
Let's add a quick note as a comment, that this doesn't need extra logic because we don't de-register events in our code today.